### PR TITLE
chore: added icons stories

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
Had to change all imports to work with this dynamic approach.
Introduces a cleaner import state regardless.

Also included one IconGallery docs
And a playground where you can see all icons in action.

Screenshot 2024-01-30 at 17 03 19
Screenshot 2024-01-30 at 17 03 09

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

File: packages/storybook/stories/Icon.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+const meta: Meta<CustomProps> = {
+  title: 'Icon',
+  component: Icon,
+  args: {
+    renderIcon: 'AiIcon',
+    size: IconSize.XXXLarge
+  },
+  argTypes: {
+    renderIcon: { control: "select", options: Object.entries(icons).map(([icon]) => icon) },
```
[the remark this answers is not in the document]
Comment (nensidosari) [reply packages/storybook/stories/Icon.stories.tsx:21]: As it also interfears a bit with saying the primary icon and secondary icon props are required, but they are empty in the storybook story.

Maybe we can do something about documenting our props more, or maybe 2 separate stories, one where we render an actual icon from what we have, another showcasing the Icon component that can take any svg as primary and secondary prop. 

Thoughts?

**Review by nensidosari — APPROVED** [review]
Love what I'm seeing 😲 🤩 

Have left 2 unblocking comments/ideas for any next PR if we would like as improvements.

File: packages/storybook/stories/Icons.mdx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { Meta, IconGallery, IconItem } from '@storybook/addon-docs';
+import * as icons from '@dailydotdev/shared/src/components/icons';
+
+<Meta title="Icon/All" />
+
+# All icons available
+
+<IconGallery>
```
[the remark this answers is not in the document]
Comment (rebelchris) [reply packages/storybook/stories/Icons.mdx:8]: Ah that's a good point!
Can probably quickly do it in one go :)

File: packages/storybook/stories/Icon.stories.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (rebelchris) [reply packages/storybook/stories/Icon.stories.tsx:21]: Yeah think indeed we should probably find a way to limit the args it renders to be more inline with the actual individual icons 😅
Was a bit hard to find a workaround for these specific "Extra" args.

Probably we can enhance the docs page here to showcase some code samples?

**Comment by rebelchris** [comment]
@nensidosari Address both your points, let me know if this is neater :)

**Review by nensidosari — APPROVED** [review]
Looks awesome with the new changes 🚀

**Pull request merged**